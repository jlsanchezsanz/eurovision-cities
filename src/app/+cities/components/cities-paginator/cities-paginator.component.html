<mat-paginator
  [length]="totalElements"
  [pageSize]="size"
  [pageSizeOptions]="pageSizeOptions"
  [pageIndex]="pageNumber"
>
</mat-paginator>
