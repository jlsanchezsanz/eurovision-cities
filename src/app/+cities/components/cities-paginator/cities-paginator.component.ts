import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-cities-paginator',
  templateUrl: './cities-paginator.component.html',
  styleUrls: ['./cities-paginator.component.scss'],
})
export class CitiesPaginatorComponent {
  @Input() public pageNumber: number;
  @Input() public pageSizeOptions: number[];
  @Input() public size: number;
  @Input() public totalElements: number;
}
