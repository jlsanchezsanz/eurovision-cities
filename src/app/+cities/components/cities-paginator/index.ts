export * from './cities-paginator.component';
