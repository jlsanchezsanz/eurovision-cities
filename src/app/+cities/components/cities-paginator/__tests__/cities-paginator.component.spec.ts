import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { MatPaginatorModule } from '@angular/material';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { CitiesPaginatorComponent } from '../cities-paginator.component';

describe('CitiesPaginatorComponent', () => {
  let component: CitiesPaginatorComponent;
  let fixture: ComponentFixture<CitiesPaginatorComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [BrowserAnimationsModule, MatPaginatorModule],
      declarations: [CitiesPaginatorComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CitiesPaginatorComponent);
    component = fixture.componentInstance;
    component.pageNumber = 3;
    component.pageSizeOptions = [5, 10, 25, 100];
    component.size = 10;
    component.totalElements = 230;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
