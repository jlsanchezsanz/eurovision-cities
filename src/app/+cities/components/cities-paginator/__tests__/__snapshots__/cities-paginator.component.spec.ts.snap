// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesPaginatorComponent should match snapshot 1`] = `
<app-cities-paginator
  pageNumber={[Function Number]}
  pageSizeOptions={[Function Array]}
  size={[Function Number]}
  totalElements={[Function Number]}
>
  <mat-paginator
    class="mat-paginator"
  >
    <div
      class="mat-paginator-outer-container"
    >
      <div
        class="mat-paginator-container"
      >
        
        <div
          class="mat-paginator-page-size"
        >
          <div
            class="mat-paginator-page-size-label"
          >
            Items per page:
          </div>
          
          <mat-form-field
            class="mat-paginator-page-size-select mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder"
          >
            <div
              class="mat-form-field-wrapper"
            >
              <div
                class="mat-form-field-flex"
              >
                
                
                <div
                  class="mat-form-field-infix"
                >
                  <mat-select
                    aria-disabled="false"
                    aria-invalid="false"
                    aria-label="Items per page:"
                    aria-multiselectable="false"
                    aria-required="false"
                    class="mat-select mat-select-empty"
                    role="listbox"
                    tabindex="0"
                  >
                    <div
                      aria-hidden="true"
                      cdk-overlay-origin=""
                      class="mat-select-trigger"
                    >
                      <div
                        class="mat-select-value"
                      >
                        
                        <span
                          class="mat-select-placeholder"
                        >
                           
                        </span>
                        
                      </div>
                      <div
                        class="mat-select-arrow-wrapper"
                      >
                        <div
                          class="mat-select-arrow"
                        />
                      </div>
                    </div>
                    
                  </mat-select>
                  <span
                    class="mat-form-field-label-wrapper"
                  >
                    
                  </span>
                </div>
                
              </div>
              
              <div
                class="mat-form-field-underline"
              >
                <span
                  class="mat-form-field-ripple"
                />
              </div>
              <div
                class="mat-form-field-subscript-wrapper"
              >
                
                
                <div
                  class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                  style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                >
                  
                  <div
                    class="mat-form-field-hint-spacer"
                  />
                </div>
              </div>
            </div>
          </mat-form-field>
          
        </div>
        <div
          class="mat-paginator-range-actions"
        >
          <div
            class="mat-paginator-range-label"
          >
            31 – 40 of 230
          </div>
          
          <button
            aria-label="Previous page"
            class="mat-paginator-navigation-previous mat-icon-button mat-button-base"
            mat-icon-button=""
            style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);"
            type="button"
          >
            <span
              class="mat-button-wrapper"
            >
              <svg
                class="mat-paginator-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
            <div
              class="mat-button-ripple mat-ripple mat-button-ripple-round"
              matripple=""
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
          <button
            aria-label="Next page"
            class="mat-paginator-navigation-next mat-icon-button mat-button-base"
            mat-icon-button=""
            style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);"
            type="button"
          >
            <span
              class="mat-button-wrapper"
            >
              <svg
                class="mat-paginator-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </span>
            <div
              class="mat-button-ripple mat-ripple mat-button-ripple-round"
              matripple=""
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
          
        </div>
      </div>
    </div>
  </mat-paginator>
</app-cities-paginator>
`;
